### PR TITLE
cake 5: Add string type hint to Driver::quote() value parameter

### PR DESCRIPTION
**Pull request by othercorey**
This matches the type hints for PDO::quote(). https://www.php.net/manual/en/pdo.quote.php

### CODE REVIEW

**Comment by othercorey** [comment]
https://github.com/phpstan/phpstan-src/pull/564

File: src/Database/Connection.php [hunk 1]
```diff
@@ -756,6 +757,9 @@ public function inTransaction(): bool
     public function quote($value, $type = 'string'): string
     {
         [$value, $type] = $this->cast($value, $type);
+        if (!is_string($value)) {
+            throw new DatabaseException('Unable to quote non-string values.');
```
Comment (othercorey) [inline src/Database/Connection.php:761]: This one might be unnecessary since it should throw a php error.

File: src/Database/Driver.php [hunk 2]
```diff
@@ -289,11 +289,11 @@ public function supportsCTEs(): bool
     /**
      * @inheritDoc
      */
-    public function quote($value, $type = PDO::PARAM_STR): string
+    public function quote(string $value, int $type = PDO::PARAM_STR): string
     {
         $this->connect();
 
-        return $this->_connection->quote((string)$value, $type);
+        return $this->_connection->quote($value, $type);
```
Comment (othercorey) [inline src/Database/Driver.php:296]: This no longer casts to string. I'm not sure we should be quoting non-string values here.

File: src/Database/Connection.php (hunk 1 above)
Comment (markstory) [reply src/Database/Connection.php]: We could skip it for now as PHP should be taking care of this.

**Pull request closed without merging**